File: yu-ai-agent-java/yu-ai-service/src/main/java/com/yuaiagent/service/chat/model/Message.java
package com.yuaiagent.service.chat.model;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 消息实体类
 */
@Data
@TableName("message")
public class Message {

    /**
     * 主键ID（自增）
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 会话ID（UUID格式）
     */
    private String chatId;

    /**
     * 角色：user | assistant | system
     */
    private String role;

    /**
     * 消息内容
     */
    private String content;

    /**
     * 消息元数据（JSON格式）
     * 包含：emotion_status, tokens_used, model, tool_calls, rag_sources
     */
    private String metadata;

    /**
     * 是否删除：0-未删除 1-已删除
     */
    @TableLogic
    private Integer isDeleted;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

}

File: yu-ai-agent-python/app/models/db/message.py
"""Message ORM Model"""
from datetime import datetime

from sqlalchemy import BigInteger, String, Text, DateTime, Boolean, JSON
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class Message(Base):
    __tablename__ = "message"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    chat_id: Mapped[str] = mapped_column(String(36), nullable=False, index=True)
    role: Mapped[str] = mapped_column(String(20), nullable=False, comment="user | assistant | system")
    content: Mapped[str] = mapped_column(Text, nullable=False)
    metadata_: Mapped[dict | None] = mapped_column("metadata", JSON, nullable=True, comment="消息元数据")
    is_deleted: Mapped[bool] = mapped_column(Boolean, default=False, comment="软删除标记")
    create_time: Mapped[datetime] = mapped_column(DateTime, default=datetime.now)
